config/config.go

```diff
@@ -849,6 +853,19 @@ func (c *ScrapeConfig) Validate(globalConfig GlobalConfig) error {
 		c.MetricNameValidationScheme = globalConfig.MetricNameValidationScheme
 	}
 
+	switch globalConfig.MetricNameEscapingScheme {
+	case model.AllowUTF8:
+		if c.MetricNameValidationScheme != UTF8ValidationConfig {
+			return errors.New("utf8 metric names requested but validation scheme is not set to UTF8")
+		}
+	case "", model.EscapeUnderscores, model.EscapeDots, model.EscapeValues:
+	default:
+		return fmt.Errorf("unknown name escaping method specified, must be one of 'allow-utf-8', 'underscores', 'dots', or 'values', got %s", globalConfig.MetricNameValidationScheme)
```

Comment:
We might want to use constants here or even create model.EscapeSchemes []string somewhere

Reply:
those are defined in Common, so that would require a change in a separate repo